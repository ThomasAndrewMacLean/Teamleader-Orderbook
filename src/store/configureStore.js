import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from './../reducers/reducers';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { routerMiddleware } from 'react-router-redux';

import createHistory from 'history/createBrowserHistory';



const logger = createLogger({
    collapsed: true
});

const history = createHistory();
const routerMw = routerMiddleware(history);

const composeEnhancers =
    typeof window === 'object' &&
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
            // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
        }) : compose;


export default function configureStore() {
    return createStore(

        rootReducer,
        composeEnhancers(
            applyMiddleware(routerMw, logger, thunk))
    );
}
