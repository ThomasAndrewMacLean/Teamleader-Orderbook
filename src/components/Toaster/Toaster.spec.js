import React from 'react';
import ReactDOM from 'react-dom';

import ConnectedToaster, { Toaster } from './Toaster';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';
import * as mocks from './../../test-helpers/mocks';
import renderer from 'react-test-renderer';


const mockStore = configureStore();

let initialState;
let store;
let wrapper;
let shallowWrapper;

const mockClearToast = jest.fn();

beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers();

    initialState = {
        data: {
            toast: {}
        }
    };
    //creates the store with any initial state or middleware needed  
    store = mockStore(initialState);
    wrapper = mount(<Provider store={store}><ConnectedToaster /></Provider>);
    shallowWrapper = shallow(<Toaster clearToast={mockClearToast} toast={mocks.getToast()} />);
});

it('snapshot', () => {
    const tree = renderer
        .create(<Provider store={store}><ConnectedToaster /></Provider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('snapshot2', () => {
    store.toast = mocks.getToast();
    const tree = renderer
        .create(<Provider store={store}><ConnectedToaster /></Provider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('should not be visible if toast msg is empty', () => {
    expect(wrapper.find('.toast').hasClass('show-toast')).toBeFalsy();
});

it('should  be visible if toast msg is not empty', () => {
    initialState = {
        data: {
            toast: mocks.getToast()
        }
    };
    store = mockStore(initialState);
    wrapper = mount(<Provider store={store}><ConnectedToaster /></Provider>);
    expect(wrapper.find('.toast').hasClass('show-toast')).toBeTruthy();
});

it('should display the message in the toast msg div', () => {
    expect(shallowWrapper.find('.toast-msg').text()).toBe('hello this is a toast');
});


it('function to clear toast runs after 3000 timeout', () => {

    let toast = { msg: 'foo', type: 'info' };
    let node = document.createElement('div');
    ReactDOM.render(<Toaster toast={toast} />, node);

    toast.msg = 'bar';
    ReactDOM.render(<Toaster toast={toast} />, node);

    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 3000);
});


it('should  have the class as defined in the type', () => {
    initialState = {
        data: {
            toast: mocks.getToast()
        }
    };
    initialState.data.toast.type = 'blablabla';
    store = mockStore(initialState);
    wrapper = mount(<Provider store={store}><ConnectedToaster /></Provider>);
    expect(wrapper.find('.toast-msg').hasClass('blablabla')).toBeTruthy();
});