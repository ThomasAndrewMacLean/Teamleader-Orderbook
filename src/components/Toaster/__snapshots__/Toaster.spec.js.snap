// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="hide-toast toast"
  onClick={[Function]}
>
  <div
    className="undefined toast-msg"
  />
</div>
`;

exports[`snapshot2 1`] = `
<div
  className="hide-toast toast"
  onClick={[Function]}
>
  <div
    className="undefined toast-msg"
  />
</div>
`;
